src/network/networkctl.c

```diff
@@ -1393,89 +1393,74 @@ static int dump_lldp_neighbors(Table *table, const char *prefix, int ifindex) {
+                JsonVariant *dhcp_server = NULL, *lease = NULL;
 
-                r = sd_bus_message_read_array(reply, 'y', &client_id, &client_id_sz);
-                if (r < 0)
-                        return bus_log_parse_error(r);
+                if (json_variant_integer(json_variant_by_key(i, "Index")) != link->ifindex)
+                        continue;
 
-                r = sd_bus_message_read_array(reply, 'y', &addr, &sz);
-                if (r < 0 || sz != 4)
-                        return bus_log_parse_error(r);
+                dhcp_server = json_variant_by_key(i, "DHCPServer");
+                if (json_variant_is_null(dhcp_server))
+                        continue;
 
-                r = sd_bus_message_read_array(reply, 'y', &gtw, &sz);
-                if (r < 0 || sz != 4)
-                        return bus_log_parse_error(r);
+                JSON_VARIANT_ARRAY_FOREACH(lease, json_variant_by_key(dhcp_server, "Leases")) {
+                        _cleanup_free_ uint8_t *client_id = NULL, *addr = NULL;
+                        _cleanup_free_ char *hostname = NULL, *id = NULL, *ip = NULL;
+                        size_t client_id_sz, addr_sz;
+                        uint64_t expiration_usec;
 
-                r = sd_bus_message_read_array(reply, 'y', &hwaddr, &sz);
-                if (r < 0)
-                        return bus_log_parse_error(r);
+                        r = json_variant_bytes(
+                                        json_variant_by_key(lease, "ClientId"), &client_id, &client_id_sz);
+                        if (r < 0)
+                                return r;
+                        r = sd_dhcp_client_id_to_string((void *) client_id, client_id_sz, &id);
+                        if (r < 0)
+                                return r;
 
-                r = sd_bus_message_read_basic(reply, 't', &expiration);
-                if (r < 0)
-                        return bus_log_parse_error(r);
+                        r = json_variant_bytes(json_variant_by_key(lease, "Address"), &addr, &addr_sz);
+                        if (r < 0)
+                                return r;
+                        if (addr_sz != sizeof(in_addr_t))
+                                return -EINVAL;
+                        r = in_addr_to_string(AF_INET, (void *) addr, &ip);
+                        if (r < 0)
+                                return r;
 
-                r = sd_dhcp_client_id_to_string(client_id, client_id_sz, &id);
-                if (r < 0)
-                        return bus_log_parse_error(r);
+                        if (json_variant_by_key(lease, "Hostname"))
+                                asprintf(&hostname,
+                                         ", \"%s\"",
+                                         json_variant_string(json_variant_by_key(lease, "Hostname")));
+                        else
+                                hostname = strdup("");
```

Comment:
Not necessary to copy an empty string.

Reply:
This was a lazy approach to deal with the free of `hostname`. But I figured out I can just pass `""` if `hostname` is NULL.